### ISSUE TITLE
FieldChoice with field 'Handshape' and machine_value 2 has many duplicates

### ISSUE THREAD

**Issue by vanlummelhuizen** [Signbank/Global-signbank#697]
@ocrasborn @Woseseltops @susanodd Do know why?

**Comment by ocrasborn** [comment]
Don't know how it came about, but remember that it was there, yes. I thought we had already polished it away. Is there perhaps an older issue about it, @vanlummelhuizen ? If it was re-introduced, we should dive into it. If not, then we should check the data, make them consistent, and remove one of the values.

**Comment by susanodd** [comment]
There's a huge amount of them now!!!

![handshapes-machine-value-2](https://user-images.githubusercontent.com/24272544/110940129-b00c4980-8336-11eb-8a46-4a7c888859da.png)

**Comment by susanodd** [comment]
I'm wondering if something is messed up in Admin.
I noticed on the other issue (field_color in admin) that Django calls the get_form multiple times when working on field choices.
I don't understand enough about how the Django automated part of Admin works to know why it is doing this.
It's also using VersionAdmin and putting things in a history.

**Comment by susanodd** [comment]
I'll try deleting all the extra ones on my server and see if they appear again.

**Comment by susanodd** [comment]
I've deleted them all, except one. That leaves the original fieldchoice duplicate machine values. In Handshapes, this is already resolved as there is only one with machine value2.

![handshapes-field-choices-2](https://user-images.githubusercontent.com/24272544/110942032-80ab0c00-8339-11eb-87c3-e282e1ffa958.png)

![handshapes-machine-value](https://user-images.githubusercontent.com/24272544/110942046-86085680-8339-11eb-9d77-e740e4376c66.png)

**Comment by susanodd** [comment]
@ocrasborn @vanlummelhuizen, shall I do this on global? There are 50 duplicates of A > T

**Comment by vanlummelhuizen** [comment]
@susanodd I  am OK with you deleting duplicates on global/live. We always have a backup. @ocrasborn agreed?

**Comment by ocrasborn** [comment]
Agreed @vanlummelhuizen @susanodd

**Comment by susanodd** [comment]
I'm working on this now. In the admin. I am getting intermittent Cannot operate on a closed database errors quite often.
I can do about 3 deletes, then this message appears. Then I wait a while, read issues in github, go back to Dictionary then Field Choices, then start deleting again.

Of the (initially 50) extra handshape field choice objects for A > T, are these fetched by queries that they are locked? @vanlummelhuizen ? I'll stop deleting for a while an you can try generating the locked database error?
Is this weird behaviour going to impact having field choices be object references in glosses? #658

**Comment by susanodd** [comment]
Here is the error message. #441 
It says something about sqlite3
@vanlummelhuizen @Woseseltops 

I can leave the browser open to the error message if you want additional information out of it.

![closed-database-top](https://user-images.githubusercontent.com/24272544/112827723-05fb2400-908f-11eb-97ab-f9ca4f913b81.png)

**Comment by vanlummelhuizen** [comment]
@susanodd These errors seem to come up when there are many things happening in the database more or less at once. @Woseseltops is trying to tackle this issue by moving to MySQL. Is it possible at all to delete a list of selected FieldChoices in the admin?

**Comment by susanodd** [comment]
Yes, I will finish deleting them. I left a few in case you wanted to generate the error message.

**Comment by susanodd** [comment]
Okay, they're deleted.

**Comment by susanodd** [comment]
I revised the tests for field choices. Since the duplicates have been deleted, it revealed a poorly coded test.

**Comment by vanlummelhuizen** [comment]
Thanks @susanodd It is already live. See also https://github.com/Signbank/Global-signbank/pull/736#issuecomment-811075242

**Comment by susanodd** [comment]
This can be closed.
